src/oatpp/core/async/CoroutineWaitList.cpp

```diff
@@ -25,16 +26,69 @@
 #include "CoroutineWaitList.hpp"
 
 #include "./Processor.hpp"
+#include <algorithm>
+#include <set>
 
 namespace oatpp { namespace async {
 
 
 CoroutineWaitList::CoroutineWaitList(CoroutineWaitList&& other) {
+  {
+    std::lock_guard<oatpp::concurrency::SpinLock> lock{other.m_lock};
     m_list = std::move(other.m_list);
+  }
+  {
+    std::lock_guard<oatpp::concurrency::SpinLock> lock{other.m_timeoutsLock};
+    m_coroutinesWithTimeout = std::move(other.m_coroutinesWithTimeout);
+    if (!m_coroutinesWithTimeout.empty()) {
+      startTimeoutCheckerThread();
+    }
+  }  
 }
 
 CoroutineWaitList::~CoroutineWaitList() {
   notifyAll();
+  m_stop = true;
+  if (m_thread.joinable()) {
+    m_thread.join();
```

Comment:
This can potentially cause a performance issue.

```cpp
class MyCoroutine ... {
  CoroutineWaitList m_waitList;
}
```

Such coroutine will block the async processor for `100ms` when destroyed.